core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -760,7 +761,10 @@ public String getPayloadAsString(String encoding) throws Exception
         String result = getPayload(DataType.STRING_DATA_TYPE, encoding);
         if (muleContext.getConfiguration().isCacheMessageAsBytes())
         {
-            cache = result.getBytes(encoding);
+            if (isValidUT8Character(result))
```

Comment:
this makes no sense. WHy to check for the first character to be utf-8 valid when you don't even know if `encoding` is utf-8 in the first place. This also seems to be fixing a symptom and not the root problem